src/privileged/fxa/api.js

```diff
@@ -32,21 +32,27 @@ function sanitizeUser(user) {
       avatarDefault = user.profileCache.profile.avatarDefault;
     }
 
-    return {
-      avatar,
-      avatarDefault,
-      email,
-      hashedUid: hashedUid(),
-      verified,
-    };
+    return getHashedUid()
+      .then((hashedUid) => {
+        return {
+          avatar,
+          avatarDefault,
+          email,
+          hashedUid,
+          verified,
+        };
+      });
   }
 
   return undefined;
 }
 
-function hashedUid() {
+async function getHashedUid() {
   try {
-    return Weave.Service.identity.hashedUID();
+    const token = await Weave.Service.identity._fetchTokenForUser();
```

Comment:
IIUC, this will make a request to the server on every call, even when it doesn't have to?

Instead, you could call `_fetchTokenForUser` only in the fallback case, if `hashedUID` throws. That would save making a roundtrip if Sync already got a token.

Reply:
That `_fetchTokenForUser` method looks like it has some side-effects. Might be a good idea to check with the folks in `#sync` that they're okay with us calling it like this, just in case.